AZURE VM " Invalid address to set: []string {\"os_profile_windows_config\", \"1534614206\", \"additional_unattend_config\"}

Hi,

I am having issues creating VM on azure using Terraform, I keep getting Invalid address to set: []string

### Terraform Version
Terraform v0.6.16

### Affected Resource(s)
resource "azurerm_virtual_machine"

### Terraform Configuration Files

```
resource "azurerm_virtual_machine" "test" {
    name = "terravm2"
    location = "West US"
    resource_group_name = "terragrp1"
    network_interface_ids = ["/subscriptions/#########################/resourceGroups/terragrp1/providers/Microsoft.Network/networkInterfaces/terranic1"]
    vm_size = "Standard_A3"

    storage_image_reference {
        publisher = "MicrosoftWindowsServer"
        offer = "WindowsServer"
        sku = "2012-R2-Datacenter"
        version = "4.0.20160518"
    }

    storage_os_disk {
        name = "myosdisk11"
        vhd_uri = "https://terrasotr1.blob.core.windows.net/vhds/myosdisk11.vhd"
        caching = "ReadWrite"
        create_option = "FromImage"
    }

    os_profile {
        computer_name = "hostname"
        admin_username = "${var.instance_admin_username}"
        admin_password = "${var.instance_admin_password}"
        }

    os_profile_windows_config {
        provision_vm_agent = true
        additional_unattend_config {
            pass = "oobeSystem"
            component = "Microsoft-Windows-Shell-Setup"
            setting_name = "FirstLogonCommands"
            content = "<FirstLogonCommands><SynchronousCommand><CommandLine>shutdown /r /t 0 /c \"initial reboot\"</CommandLine><Description>reboot</Description><Order>1</Order></SynchronousCommand></FirstLogonCommands>"
        }
    }
}
```

### Debug Output
2016/06/07 12:44:13 [ERROR] root: eval: *terraform.EvalRefresh, err: azurerm_virtual_machine.test: [DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: &errors.errorString{s:"Invalid address to set: []string{\"os_profile_windows_config\", \"1534614206\", \"additional_unattend_config\"}"}
2016/06/07 12:44:13 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_virtual_machine.test: [DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: &errors.errorString{s:"Invalid address to set: []string{\"os_profile_windows_config\", \"1534614206\", \"additional_unattend_config\"}"}
2016/06/07 12:44:13 [ERROR] root: eval: *terraform.EvalOpFilter, err: azurerm_virtual_machine.test: [DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: &errors.errorString{s:"Invalid address to set: []string{\"os_profile_windows_config\", \"1534614206\", \"additional_unattend_config\"}"}
2016/06/07 12:44:13 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_virtual_machine.test: [DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: &errors.errorString{s:"Invalid address to set: []string{\"os_profile_windows_config\", \"1534614206\", \"additional_unattend_config\"}"}

* azurerm_virtual_machine.test: [DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: &errors.errorString{s:"Invalid address to set: []string{\"os_profile_windows_config\", \"1534614206\", \"additional_unattend_config\"}"}

--- a/builtin/providers/azurerm/resource_arm_virtual_machine.go
+++ b/builtin/providers/azurerm/resource_arm_virtual_machine.go
@@ -257,7 +257,7 @@ func resourceArmVirtualMachine() *schema.Resource {
 							},
 						},
 						"additional_unattend_config": {
-							Type:     schema.TypeSet,
+							Type:     schema.TypeList,
 							Optional: true,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
@@ -506,7 +506,7 @@ func resourceArmVirtualMachineRead(d *schema.ResourceData, meta interface{}) err
 	}
 
 	if resp.Properties.OsProfile.WindowsConfiguration != nil {
-		if err := d.Set("os_profile_windows_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileWindowsConfigHash, flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(resp.Properties.OsProfile.WindowsConfiguration))); err != nil {
+		if err := d.Set("os_profile_windows_config", flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(resp.Properties.OsProfile.WindowsConfiguration)); err != nil {
 			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: %#v", err)
 		}
 	}
@@ -980,7 +980,7 @@ func expandAzureRmVirtualMachineOsProfileWindowsConfig(d *schema.ResourceData) (
 		}
 	}
 	if v := osProfileConfig["additional_unattend_config"]; v != nil {
-		additionalConfig := v.(*schema.Set).List()
+		additionalConfig := v.([]interface{})
 		if len(additionalConfig) > 0 {
 			additionalConfigContent := make([]compute.AdditionalUnattendContent, 0, len(additionalConfig))
 			for _, addConfig := range additionalConfig {

--- a/builtin/providers/azurerm/resource_arm_virtual_machine_test.go
+++ b/builtin/providers/azurerm/resource_arm_virtual_machine_test.go
@@ -113,6 +113,24 @@ func TestAccAzureRMVirtualMachine_basicWindowsMachine(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMVirtualMachine_windowsUnattendedConfig(t *testing.T) {
+	ri := acctest.RandInt()
+	config := fmt.Sprintf(testAccAzureRMVirtualMachine_windowsUnattendedConfig, ri, ri, ri, ri, ri, ri)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineExists("azurerm_virtual_machine.test"),
+				),
+			},
+		},
+	})
+}
+
 func testCheckAzureRMVirtualMachineExists(name string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		// Ensure we have enough information in state to look up in API
@@ -505,6 +523,95 @@ resource "azurerm_virtual_machine" "test" {
     os_profile_windows_config {
 	enable_automatic_upgrades = false
 	provision_vm_agent = true
+    }
+}
+`
+
+var testAccAzureRMVirtualMachine_windowsUnattendedConfig = `
+resource "azurerm_resource_group" "test" {
+    name = "acctestrg-%d"
+    location = "West US"
+}
+
+resource "azurerm_virtual_network" "test" {
+    name = "acctvn-%d"
+    address_space = ["10.0.0.0/16"]
+    location = "West US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_subnet" "test" {
+    name = "acctsub-%d"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    virtual_network_name = "${azurerm_virtual_network.test.name}"
+    address_prefix = "10.0.2.0/24"
+}
+
+resource "azurerm_network_interface" "test" {
+    name = "acctni-%d"
+    location = "West US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+
+    ip_configuration {
+    	name = "testconfiguration1"
+    	subnet_id = "${azurerm_subnet.test.id}"
+    	private_ip_address_allocation = "dynamic"
+    }
+}
+
+resource "azurerm_storage_account" "test" {
+    name = "accsa%d"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    location = "westus"
+    account_type = "Standard_LRS"
+
+    tags {
+        environment = "staging"
+    }
+}
+
+resource "azurerm_storage_container" "test" {
+    name = "vhds"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    storage_account_name = "${azurerm_storage_account.test.name}"
+    container_access_type = "private"
+}
+
+resource "azurerm_virtual_machine" "test" {
+    name = "acctvm-%d"
+    location = "West US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    network_interface_ids = ["${azurerm_network_interface.test.id}"]
+    vm_size = "Standard_A0"
+
+    storage_image_reference {
+	publisher = "MicrosoftWindowsServer"
+	offer = "WindowsServer"
+	sku = "2012-Datacenter"
+	version = "latest"
+    }
+
+    storage_os_disk {
+        name = "myosdisk1"
+        vhd_uri = "${azurerm_storage_account.test.primary_blob_endpoint}${azurerm_storage_container.test.name}/myosdisk1.vhd"
+        caching = "ReadWrite"
+        create_option = "FromImage"
+    }
+
+    os_profile {
+	computer_name = "winhost01"
+	admin_username = "testadmin"
+	admin_password = "Password1234!"
+    }
+
+    os_profile_windows_config {
+        provision_vm_agent = true
+        additional_unattend_config {
+            pass = "oobeSystem"
+            component = "Microsoft-Windows-Shell-Setup"
+            setting_name = "FirstLogonCommands"
+            content = "<FirstLogonCommands><SynchronousCommand><CommandLine>shutdown /r /t 0 /c \"initial reboot\"</CommandLine><Description>reboot</Description><Order>1</Order></SynchronousCommand></FirstLogonCommands>"
+        }
     }
 }
 `

